### PR TITLE
Db

### PR DESCRIPTION
**Pull request by 5t4mp5**
Added User, LineItem, Cart models

TODO Improve validations and custom errors. Improve seed file.

### CODE REVIEW

File: db/models/Cart.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const { Sequelize } = db;
+
+const Cart = db.define("cart", {
+  id: {
+    type: Sequelize.UUID,
+    defaultValue: Sequelize.UUIDV4,
+    primaryKey: true
+  },
+  status: {
+    type: Sequelize.STRING,
```
Comment (chem2016) [inline db/models/Cart.js:11]: Should we try to use Sequelize.ENUM(Sequelize.STRING)?

File: db/models/index.js [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
 
-module.exports = { Product, Category };
+LineItem.belongsTo(Product);
+Product.hasMany(LineItem);
+
+LineItem.belongsTo(Cart);
+Cart.hasMany(LineItem);
+
+Cart.belongsTo(User);
+User.hasMany(Cart);
```
Comment (chem2016) [inline db/models/index.js]: Should one user only have one cart? The cart can hold many products and lineItems?

**Review by chem2016 — COMMENTED** [review]
@5t4mp5 Everything looks good, I just have two comments please read them, thanks!

File: db/models/Cart.js (hunk 1 above)
Comment (5t4mp5) [reply db/models/Cart.js:11]: I've been advised by a Sam, and whoever graded my senior checkpoint not to use Enum. ENUM fields do not get dropped when you sync the db. That would make any changes to that field problematic to deal with. I can change it if you like, but I'd still use the validation to throw a custom error.

File: db/models/index.js (hunk 2 above)
Comment (5t4mp5) [reply db/models/index.js]: Good catch. I changed that relationship.

**Review by chem2016 — APPROVED** [review]
Thank you for the changes!

**Pull request merged**